### PR TITLE
speed-dreams: New version 2.2.3

### PR DESCRIPTION
**Pull request by BeagleJoe**
Crashes at program exit.
Needs Mesa updated.

### CODE REVIEW

File: games-sports/speed-dreams/speed_dreams-2.2.3.recipe [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+DESCRIPTION="Speed Dreams is a 3d cross-platform, open source motorsport simulation and racing game.
+
+Speed Dreams is a fork of the open racing car simulator Torcs, aiming to implement exciting new \
+features, cars, tracks and AI opponents to make a more enjoyable game for the player, as well as \
+constantly improving visual and physics realism."
+HOMEPAGE="http://www.speed-dreams.org/"
+COPYRIGHT="2009-2018 Jean-Philippe Meuret"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="svn://svn.code.sf.net/p/speed-dreams/code/tags/2.2.3"
```
Comment (korli) [inline games-sports/speed-dreams/speed_dreams-2.2.3.recipe:11]: this is overwritten two lines later

File: games-sports/speed-dreams/speed_dreams-2.2.3.recipe [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+SOURCE_FILENAME="sd-$portVersion"
```
Comment (korli) [inline games-sports/speed-dreams/speed_dreams-2.2.3.recipe:12]: SOURCE_FILENAME isn't needed

File: games-sports/speed-dreams/additional-files/speed_dreams.rdef [hunk 3]
```diff
@@ -0,0 +1,87 @@
+resource app_flags B_SINGLE_LAUNCH | B_ARGV_ONLY;
+
+resource app_version {
+		major = 2,
```
Comment (korli) [inline games-sports/speed-dreams/additional-files/speed_dreams.rdef:4]: you could avoid hardcoding the version here.
Please check https://github.com/haikuports/haikuports/blob/master/app-backup/luckybackup/luckybackup-0.5.0.recipe#L85 for an example

File: games-sports/speed-dreams/speed_dreams-2.2.3.recipe (hunk 1 above)
Comment (BeagleJoe) [reply games-sports/speed-dreams/speed_dreams-2.2.3.recipe]: Fixed, thanks.

File: games-sports/speed-dreams/speed_dreams-2.2.3.recipe (hunk 2 above)
Comment (BeagleJoe) [reply games-sports/speed-dreams/speed_dreams-2.2.3.recipe]: Fixed, thanks.

File: games-sports/speed-dreams/additional-files/speed_dreams.rdef (hunk 3 above)
Comment (BeagleJoe) [reply games-sports/speed-dreams/additional-files/speed_dreams.rdef:4]: Thanks for the example. I'm looking into possibly using CMake to generate the rdef file.

Comment (BeagleJoe) [reply games-sports/speed-dreams/additional-files/speed_dreams.rdef]: Done, thanks.

**Comment by korli** [comment]
Looks good, can merge as disabled, right?

**Comment by BeagleJoe** [comment]
> Looks good, can merge as disabled, right?

Yes, thanks!

File: games-sports/speed-dreams/speed_dreams-2.2.3.recipe [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	lib:libcurl$secondaryArchSuffix
+	lib:libenet$secondaryArchSuffix
+	lib:libexpat$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libGLU$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix
+	lib:libogg$secondaryArchSuffix
+	lib:libopenal$secondaryArchSuffix
+	lib:libpng16$secondaryArchSuffix
+	lib:libsdl$secondaryArchSuffix
```
Comment (Begasus) [inline games-sports/speed-dreams/speed_dreams-2.2.3.recipe:35]:
```diff
- 	lib:libsdl$secondaryArchSuffix
+ 	lib:libSDL_1.2$secondaryArchSuffix
```

File: games-sports/speed-dreams/speed_dreams-2.2.3.recipe [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+	devel:libenet$secondaryArchSuffix
+	devel:libexpat$secondaryArchSuffix
+	devel:libGL$secondaryArchSuffix
+	devel:libGLU$secondaryArchSuffix
+	devel:libjpeg$secondaryArchSuffix
+	devel:libogg$secondaryArchSuffix
+	devel:libopenal$secondaryArchSuffix
+	devel:libplibnet$secondaryArchSuffix
+	devel:libpng$secondaryArchSuffix
+	devel:libsdl$secondaryArchSuffix
```
Comment (Begasus) [inline games-sports/speed-dreams/speed_dreams-2.2.3.recipe:55]:
```diff
-	devel:libpng$secondaryArchSuffix
-	devel:libsdl$secondaryArchSuffix
+	devel:libpng16$secondaryArchSuffix
+	devel:libSDL_1.2$secondaryArchSuffix
```

Comment (Begasus) [reply games-sports/speed-dreams/speed_dreams-2.2.3.recipe:55]: Issue for those are mentioned in: https://github.com/haikuports/haikuports/issues/3878 and https://github.com/haikuports/haikuports/issues/3779

**Review by Begasus — CHANGES_REQUESTED** [review]

**Comment by Begasus** [comment]
> Should the ARCHITECTURES be changed from ? to !

No that's OK, `?` means disabled, `!` means broken :)

**Comment by BeagleJoe** [comment]
I made the requested changes.
I also was able to patch the game to use SDL2 instead of SDL1.
For some reason the arrow keys did not work with SDL1.
I used lib:libSDL2_2.0secondaryArchSuffix and devel:libSDL2_2.0secondaryArchSuffix
I think this is correct.

File: games-sports/speed-dreams/speed_dreams-2.2.3.recipe [hunk 6]
```diff
@@ -0,0 +1,102 @@
+SUMMARY="Open Motorsport Simulator"
+DESCRIPTION="Speed Dreams is a 3d cross-platform, open source motorsport simulation and racing game.
+
+Speed Dreams is a fork of the open racing car simulator Torcs, aiming to implement exciting new \
+features, cars, tracks and AI opponents to make a more enjoyable game for the player, as well as \
+constantly improving visual and physics realism."
```
Comment (Begasus) [inline games-sports/speed-dreams/speed_dreams-2.2.3.recipe:6]: Keep the lines to a max of 80 characters (I know it's a left over from previous versions) :)
See: https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#description

File: games-sports/speed-dreams/speed_dreams-2.2.3.recipe [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+HOMEPAGE="http://www.speed-dreams.org/"
+COPYRIGHT="2009-2018 Jean-Philippe Meuret"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://sourceforge.net/projects/speed-dreams/files/$portVersion/speed-dreams-src-base-$portVersion-r7616.tar.xz"
+CHECKSUM_SHA256="c67e48a083cd86f6c0062f0b38ed91a5d25274ea8aa656ee182f3f5074e96ff0"
+SOURCE_DIR=""
+ADDITIONAL_FILES="speed_dreams.rdef.in"
+PATCHES="speed_dreams-$portVersion.patchset"
```
Comment (Begasus) [inline games-sports/speed-dreams/speed_dreams-2.2.3.recipe:15]: Swap these 2 lines, PATCHES come before ADDITIONAL_FILES
See: https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#ordering

File: games-sports/speed-dreams/speed_dreams-2.2.3.recipe [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+Speed Dreams is a fork of the open racing car simulator Torcs, aiming to implement exciting new \
+features, cars, tracks and AI opponents to make a more enjoyable game for the player, as well as \
+constantly improving visual and physics realism."
+HOMEPAGE="http://www.speed-dreams.org/"
+COPYRIGHT="2009-2018 Jean-Philippe Meuret"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://sourceforge.net/projects/speed-dreams/files/$portVersion/speed-dreams-src-base-$portVersion-r7616.tar.xz"
+CHECKSUM_SHA256="c67e48a083cd86f6c0062f0b38ed91a5d25274ea8aa656ee182f3f5074e96ff0"
+SOURCE_DIR=""
```
Comment (Begasus) [inline games-sports/speed-dreams/speed_dreams-2.2.3.recipe:13]: Not needed:
```diff
- SOURCE_DIR=""
```

File: games-sports/speed-dreams/speed_dreams-2.2.3.recipe [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+	lib:libcurl$secondaryArchSuffix
+	lib:libenet$secondaryArchSuffix
+	lib:libexpat$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libGLU$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix
+	lib:libogg$secondaryArchSuffix
+	lib:libopenal$secondaryArchSuffix
+	lib:libpng16$secondaryArchSuffix
+#	lib:libSDL_1.2$secondaryArchSuffix
```
Comment (Begasus) [inline games-sports/speed-dreams/speed_dreams-2.2.3.recipe:35]: If libSDL_1.2 isn't used it can be removed here and the line for devel:libSDL_1.2

**Review by Begasus — COMMENTED** [review]
Some of the comments are things that needed a fix in previous version also, I'm guessing we were not that strict back then :)

**Review by Begasus — APPROVED** [review]

**Comment by Begasus** [comment]
Could you remove the previous recipes? That way we can check up on the changes made here.
I also saw that some other OS's use the game data? (didn't check repology in the past I guess) :)

Gentoo: https://gitweb.gentoo.org/repo/gentoo.git/tree/games-sports/speed-dreams/speed-dreams-2.2.3.ebuild
openSUSE: https://build.opensuse.org/package/show/games/speed-dreams
Repology references: https://repology.org/project/speed-dreams/versions

File: games-sports/speed-dreams/speed_dreams-2.2.3.recipe (hunk 6 above)
Comment (BeagleJoe) [reply games-sports/speed-dreams/speed_dreams-2.2.3.recipe:6]: I fixed up the description, but I was not sure about breaking up SOURCE_URI, CHECKSUM_SHA256 to less than 80 lines?

File: games-sports/speed-dreams/speed_dreams-2.2.3.recipe (hunk 8 above)
Comment (BeagleJoe) [reply games-sports/speed-dreams/speed_dreams-2.2.3.recipe:13]: Without that line,:
`Error: speed_dreams-2.2.3 doesn't exist in sources! Define SOURCE_DIR in recipe?`

**Comment by BeagleJoe** [comment]
In general, do you prefer small commits, or resolve all the issues and then do one commit?

> Could you remove the previous recipes? 

Do you mean the ones for 2.0.0 and 2.2.2 ?

> I also saw that some other OS's use the game data?

Yes, they are additional cars and tracks. They are also quite large. The full AppImage at SourceForge is 1.8GB.
What we have with the current recipe is the base game.

File: games-sports/speed-dreams/speed_dreams-2.2.3.recipe (hunk 8 above)
Comment (Begasus) [reply games-sports/speed-dreams/speed_dreams-2.2.3.recipe:13]: Really should check a build, will get back on it :)

**Comment by Begasus** [comment]
Better to keep it to the base package for now then

File: games-sports/speed-dreams/speed_dreams-2.2.3.recipe (hunk 6 above)
Comment (Begasus) [reply games-sports/speed-dreams/speed_dreams-2.2.3.recipe:6]: Only the DESCRIPTION is involved here, the others are OK

File: games-sports/speed-dreams/speed_dreams-2.2.3.recipe (hunk 9 above)
Comment (BeagleJoe) [reply games-sports/speed-dreams/speed_dreams-2.2.3.recipe]: Removed

File: games-sports/speed-dreams/speed_dreams-2.2.3.recipe (hunk 7 above)
Comment (BeagleJoe) [reply games-sports/speed-dreams/speed_dreams-2.2.3.recipe]: Done

File: games-sports/speed-dreams/speed_dreams-2.2.3.recipe (hunk 6 above)
Comment (BeagleJoe) [reply games-sports/speed-dreams/speed_dreams-2.2.3.recipe]: Fixed Description

File: games-sports/speed-dreams/speed_dreams-2.2.3.recipe [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+REVISION="1"
+SOURCE_URI="https://sourceforge.net/projects/speed-dreams/files/$portVersion/speed-dreams-src-base-$portVersion-r7616.tar.xz"
+
+CHECKSUM_SHA256="c67e48a083cd86f6c0062f0b38ed91a5d25274ea8aa656ee182f3f5074e96ff0"
+SOURCE_DIR=""
+
+PATCHES="speed_dreams-$portVersion.patchset"
+ADDITIONAL_FILES="speed_dreams.rdef.in"
+
+ARCHITECTURES="!x86_gcc2 ?x86 ?x86_64"
```
Comment (Begasus) [inline games-sports/speed-dreams/speed_dreams-2.2.3.recipe:17]: Could you update this to the latest change at haikuporter?
```diff
- ARCHITECTURES="!x86_gcc2 ?x86 ?x86_64"
+ ARCHITECTURES="?all !x86_gcc2"
```

**Pull request merged**